### DISCUSSION TITLE
Does Connectivity.ConnectivityChanged detect internet changes even without connection lose?

### DISCUSSION THREAD

**Discussion by darren-kay** [dotnet/maui#21017] [Q&A] [answered]
Is Connectivity.ConnectivityChanged triggered just by connection service or also trigger is the wifi lost access to the internet?

For example, if the ISP for a Wifi network had a service interruption could I detect this using the Connectivity.ConnectivityChanged event?

**Comment by ghost** [comment, marked as answer]
No, it does not trigger when the wifi lost access to the internet.
I have used Android, set up WiFi access point, build the app, connected this phone to the WiFi access point and then turn off the mobile data which (internet access). The connectivityChanged does not trigger.

You can read about this in the [Limitation section of the documentation](https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/communication/networking?view=net-maui-9.0&tabs=android#limitations).

You could have a timer, for example, every 6 seconds, checking if the Internet is available. [How to use Connectivity](Connectivity.Current.NetworkAccess)?
